state-sync/diem-data-client/src/diemnet/tests.rs

```diff
@@ -175,6 +204,171 @@ async fn test_request_works_only_when_data_available() {
         .get_transactions_with_proof(100, 50, 100, false)
         .await
         .unwrap();
+    assert_eq!(response.payload, TransactionListWithProof::new_empty());
+}
+
+// 1. 2 peers
+// 2. one advertises bad range, one advertises honest range
+// 3. sending a bunch of requests to the bad range (which will always go to the
+//    bad peer) should lower bad peer's score
+// 4. eventually bad peer score should hit threshold and we err with no available
 
-    assert_eq!(response.payload, TransactionListWithProof::new_empty(),);
+#[tokio::test]
+async fn bad_peer_is_eventually_banned_internal() {
```

Comment:
nit: there is a lot of code duplication between these two tests. But, I recognize that these style of refactors are probably premature. Feel free to ignore.